string ADD=ADD
string REMOVE=REMOVE
string CLEAR=CLEAR

string action


string[] frames_in_contact

